src/fileoverview_comment_transformer.ts

```diff
@@ -75,20 +75,48 @@ function augmentFileoverviewComments(tags: jsdoc.Tag[]) {
     (sf: ts.SourceFile) => ts.SourceFile {
   return (sf: ts.SourceFile) => {
-    let comments: ts.SynthesizedComment[] = [];
-    // Use trailing comments because that's what transformer_util.ts creates (i.e. by convention).
-    if (sf.statements.length && sf.statements[0].kind === ts.SyntaxKind.NotEmittedStatement) {
-      comments = ts.getSyntheticTrailingComments(sf.statements[0]) || [];
+    const text = sf.getFullText();
+
+    let fileComments: ts.SynthesizedComment[] = [];
+    const firstStatement = sf.statements.length && sf.statements[0] || null;
+
+    const originalComments = ts.getLeadingCommentRanges(text, 0) || [];
+    if (!firstStatement) {
+      // In an empty source file, all comments are file-level comments.
+      fileComments = synthesizeCommentRanges(sf, originalComments);
+    } else {
+      // Search for the first comment split from the file with a \n\n. All comments before that are
+      // considered fileoverview comments, all comments after that belong to the next statement(s).
+      // If none found, comments remains empty, and the code below will insert a new fileoverview
+      // comment.
+      for (let i = 0; i < originalComments.length; i++) {
+        const end = originalComments[i].end;
+        if (!text.substring(end).startsWith('\n\n')) continue;
+        // This comment is separated from the source file with a double break, marking it (and any
+        // preceding comments) as a file-level comment. Split them off and attach them onto a
+        // NotEmittedStatement, so that they do not get lost later on.
+        const synthesizedComments = jsdoc.synthesizeLeadingComments(firstStatement);
+        const notEmitted = ts.createNotEmittedStatement(sf);
+        // Modify the comments on the firstStatement in place by removing the file-level comments.
+        fileComments = synthesizedComments.splice(0, i + 1);
+        // Move the fileComments onto notEmitted.
+        ts.setSyntheticLeadingComments(notEmitted, fileComments);
+        // TODO(martinprobst): consider checking here whether any of the trailing comments contains
+        // an @fileoverview etc, and reporting an error if so.
+        sf = updateSourceFileNode(
+            sf, ts.createNodeArray([notEmitted, firstStatement, ...sf.statements.slice(1)]));
+        break;
+      }
     }
 
-    // Closure Compiler considers the *last* comment with @fileoverview (or @externs or @nocompile)
-    // that has not been attached to some other tree node to be the file overview comment, and
-    // only applies @suppress tags from it.
-    // AJD considers *any* comment mentioning @fileoverview.
+    // Closure Compiler considers the *last* comment with @fileoverview (or @externs or
+    // @nocompile) that has not been attached to some other tree node to be the file overview
+    // comment, and only applies @suppress tags from it. AJD considers *any* comment mentioning
+    // @fileoverview.
```

Comment:
Looks like maybe you're rewrapping to 80.  Maybe it's worth doing that in a pre-pass?